web/client/components/mapcontrols/search/SearchBar.jsx

```diff
@@ -405,13 +464,22 @@ class SearchBar extends React.Component {
         this.props.onSearchReset();
     };
 
+    clearCoordinates = () => {
+        this.props.onClearCoordinatesSearch({owner: "search"});
+        this.props.onChangeCoord("lat", "");
+        this.props.onChangeCoord("lon", "");
+
+    };
+
     changeCoord = (coord, value) => {
-        let val = isNaN(parseFloat(value)) ? undefined : parseFloat(value);
-        this.setState(() => ({
-            [coord]: val
-        }));
+        let val = isNaN(parseFloat(value)) ? "" : parseFloat(value);
+
+        this.props.onChangeCoord(coord, val);
+        if (!this.areValidCoordinates()) {
+            this.props.onClearCoordinatesSearch({owner: "search"});
+        }
     }
-    areValidCoordinates = () => (isNumber(this.state.lon) && isNumber(this.state.lat))
+    areValidCoordinates = () => (isNumber(this.props.coordinate.lon) && isNumber(this.props.coordinate.lat))
```

Comment:
You can eventually check that coordinates are in the valid range for lat and lon (not mandatory, just a suggestion)

Reply:
the range validation is demanded to the coordinate editor components and is performed using enhancers (AeronauticalCoordinateEditor) or as part of the component (DecimalCoordinateEditor) and the **contraints** prop

here i'm just checking the presence of two number in order to pass the values to the zoom actions, state. etc.

Reply:
ok, fine